### ISSUE TITLE
downsampling continuous query "time clause" error

### ISSUE THREAD

**Issue by nicksellen** [influxdb/influxdb#3117]
I installed influxdb on my Ubuntu 15.04 machine from https://s3.amazonaws.com/influxdb/influxdb_latest_amd64.deb

## Versions

```
$ /opt/influxdb/influxd version
InfluxDB v0.9.0 (git: 471117ce5b308e59f8f247d8a88a52ead553b602)
$ /opt/influxdb/influx version
Connected to http://localhost:8086 version FIXME
InfluxDB shell 0.9.0
$ grep DESC /etc/lsb-release 
DISTRIB_DESCRIPTION="Ubuntu 15.04"
$ uname -a
Linux zoya 3.19.0-18-generic #18-Ubuntu SMP Tue May 19 18:31:35 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux
```

(`/opt/influxdb/influx version` has `FIXME` in version output, oops!)

## Recreate issue

```
# create mydb
curl -G http://localhost:8086/query --data-urlencode "q=CREATE DATABASE mydb"

# write some data
curl -i -XPOST 'http://localhost:8086/write?db=mydb&rp=default&precision=s' -d 'cpu_load_short,host=server01,region=us-west value=0.5 1435143600
cpu_load_short,host=server01,region=us-west value=1.0 1435143601
cpu_load_short,host=server01,region=us-west value=0.7 1435143602'

# create continuous query
curl -G http://localhost:8086/query --data-urlencode 'q=CREATE CONTINUOUS QUERY cpu_load_5m ON mydb BEGIN
  SELECT MEAN(value) as value INTO cpu_load_5m FROM cpu_load_short GROUP BY time(5m) 
END'
```

The log shows the query is missing the time clause:
```
[continuous_querier] 2015/06/24 13:03:25 error executing query: SELECT mean(value) AS value INTO "mydb"."default".cpu_load_5m FROM "mydb"."default".cpu_load_short GROUP BY time(5m): err = aggregate functions with GROUP BY time require a WHERE time clause
```

Seems the `tr == targetNotRequired` condition is passing when it shouldn't in https://github.com/influxdb/influxdb/blob/master/influxql/ast.go#L979.

Or am I doing something wrong?

**Comment by otoolep** [comment]
I believe this is will be fixed in 0.9.1.

@dgnorton

**Comment by beckettsean** [comment]
Fixed in https://github.com/influxdb/influxdb/pull/2985, which is on master and will be in 0.9.1
